Dynatrace/dynatrace-operator
# Get http proxy in OneAgent from mounted secret instead of env variable

## Description

Remove http proxy env variable in os-OneAgent and mount a secret with its value instead (in the same fashion as ActiveGate does, actually using the same secret for both now).

## How can this be tested?

Apply a Dynakube with proxy (either with `value` or `valueFrom`) and check that OneAgent gets a mounted volume on the expected path (`/var/lib/dynatrace/secrets/internal-proxy`). Also, you can check its value in the container shell. (see pictures)

![image](https://github.com/Dynatrace/dynatrace-operator/assets/61692514/6c6aaa26-dbbd-45c5-97cc-55d6e310a637)
![image](https://github.com/Dynatrace/dynatrace-operator/assets/61692514/ed3f4687-854b-461e-a821-3b6796b0339f)

Check also that the reconciliator creating the proxy secret works, it should create `<dynakubeName>-internal-proxy`.

![image](https://github.com/Dynatrace/dynatrace-operator/assets/61692514/3388723d-a38e-4c6e-95b6-a2aab57700d7)

## Checklist
